src/app/docs/kagent/operations/operational-considerations/page.mdx

```diff
@@ -0,0 +1,137 @@
+---
+title: "Operational considerations"
+pageOrder: 1
+description: "Important operational considerations when running kagent in production."
+---
+
+export const metadata = {
+  title: "Operational Considerations",
+  description: "Important operational considerations when running kagent in production.",
+  author: "kagent.dev"
+};
+
+# Operational considerations
+
+Review the following operational considerations when running kagent in production environments, including database configuration, high availability, and secret management.
+
+## Automatic agent restart on secret updates
+
+Kagent automatically restarts agents when you update the secrets that the agents reference. This restart ensures that agents pick up new API keys, TLS certificates, and other secret values without manual intervention.
+
+The following secret updates trigger automatic agent restarts:
+
+- **API keys**: Secrets referenced in `ModelConfig` resources (e.g., `OPENAI_API_KEY`, `ANTHROPIC_API_KEY`)
+- **TLS certificates**: Secrets referenced in `ModelConfig` TLS configuration (e.g., CA certificates)
+- **Environment variables**: Any secrets referenced via `secretKeyRef` in agent deployment specifications
+
+## Leader election when controller is scaled
+
+When you scale the kagent controller to multiple replicas for high availability, leader election is automatically enabled. This ensures that only one controller instance actively reconciles resources at a time, preventing conflicts and duplicate operations.
+
+### Leader election scenarios
+
+- **Single replica**: No leader election needed; the single controller instance handles all operations
+- **Multiple replicas**: Leader election is automatically enabled when `controller.replicas > 1`
+- **Active leader**: Only the elected leader performs reconciliation operations
+- **Standby replicas**: Other replicas remain ready but do not perform reconciliation until they become the leader
```

Comment:
maybe add periods at the end since theyre complete sentences
```suggestion
- **Single replica**: No leader election needed; the single controller instance handles all operations.
- **Multiple replicas**: Leader election is automatically enabled when `controller.replicas > 1`.
- **Active leader**: Only the elected leader performs reconciliation operations.
- **Standby replicas**: Other replicas remain ready but do not perform reconciliation until they become the leader.
```